src/commands/mapsheet-coverage/mapsheet.coverage.ts

```diff
@@ -0,0 +1,325 @@
+    });
+
+    await fsa.write(fsa.join(OutputPath, 'file-list.json'), JSON.stringify(todo));
+    logger.info(
+      {
+        duration: performance.now() - startTime,
+        layersFound: allLayers.features.length,
+        layersNeeded: layersRequired.features.length,
+        itemsToCreate: todo.length,
+      },
+      'MapSheetCoverage:Done',
+    );
+  },
+});
+
+async function compareCreation(
+  compareLocation: string,
+  mapSheets: Map<string, string[]>,
+  hashQueueLength = 25,
+): Promise<string[]> {
+  logger.info({ compareTo: compareLocation, mapSheetCount: mapSheets.size }, 'MapSheet:Compare');
+
+  // Limit the number of files hashing concurrently
+  const hashQueue = pLimit(hashQueueLength);
+
+  // Joining STAC document locations as file paths can be tricky, use the built in URL lib to handle the joins
+  const compareUrl = tryParseUrl(compareLocation);
+
+  const collectionJson = await fsa.readJson<StacCollection>(compareLocation);
+
+  // List of mapsheets that are the same as the compare location
+  const sheetsToSkip: string[] = [];
+
+  for (const [sheetCode, sourceFiles] of mapSheets) {
+    const itemLink = collectionJson.links.find((f) => f.href.endsWith(sheetCode + '.json'));
+
+    // Mapsheet does not exist in current collection json, it is new file to be created
+    if (itemLink == null) {
+      logger.info({ sheetCode: sheetCode, sourceFiles: sourceFiles.length }, 'MapSheet:Compare:New');
+      continue;
+    }
+    const itemJson = await fsa.readJson<StacItem>(urlToString(new URL(itemLink.href, compareUrl)));
+
+    const derivedFrom = itemJson.links.filter((f) => f.rel === 'derived_from');
+    // Difference in the number of files needed to create this mapsheet, so it needs to be recreated
+    if (derivedFrom.length !== sourceFiles.length) {
+      logger.debug(
+        { sheetCode: sheetCode, sourceLocations: sourceFiles, oldLocations: derivedFrom.map((m) => m.href) },
+        'mapsheet:difference',
+      );
+      continue;
+    }
+
+    let needsToBeCreated = false;
+    await Promise.all(
+      derivedFrom.map(async (item, index) => {
+        // A difference has already been found skip checking the rest of the checksums
+        if (needsToBeCreated) return;
+
+        const sourceFile = sourceFiles[index];
+        if (sourceFile == null || item.href !== sourceFile.replace('.tiff', '.json')) {
+          logger.debug({ sheetCode, source: item.href }, 'MapSheet:Compare:source-difference');
+          needsToBeCreated = true;
+          return;
+        }
+
+        // No checksum found in source collection link, force re-create the file
+        if (item['file:checksum'] == null) {
+          logger.warn({ sheetCode, source: item.href }, 'MapSheet:Compare:source-checksum-missing');
+          needsToBeCreated = true;
+          return;
+        }
+
+        // TODO: to improve performance further we could use the source collection.json as it contains all the item checksums
```

Comment:
Unless we want to track the differences?

Reply:
the collection.json has all the checksums for all the item jsons, so would just fetch one file vs fetching 100s